app/utils/commands.js

```diff
@@ -0,0 +1,253 @@
+    withValue: false
+  },
+  duration: {
+    long: '--duration',
+    short: '-d',
+    description: 'Specify duration for pausing breaks (Pause only) [infinitely|until-morning|seconds (as number)]',
+    withValue: true
+  }
+}
+
+const allCommands = {
+  help: {
+    description: 'Show this help message'
+  },
+  version: {
+    description: 'Show current stretchly version'
+  },
+  reset: {
+    description: 'Reset breaks'
+  },
+  pause: {
+    description: 'Pause breaks',
+    options: [allOptions.duration]
+  },
+  resume: {
+    description: 'Resume from a pause'
+  },
+  toggle: {
+    description: 'Toggle breaks between resume/paused'
+  },
+  mini: {
+    description: 'Skips to next Mini Break',
+    options: [allOptions.title, allOptions.skip]
+  },
+  long: {
+    description: 'Skips to next Long Break',
+    options: [allOptions.text, allOptions.title, allOptions.skip]
+  }
+}
+
+const allExamples = [{
+  cmd: 'stretchly pause',
+  description: 'Pause breaks indefinitly'
+},
+{
+  cmd: 'stretchly pause -d 3600',
+  description: 'Pause breaks for one hour'
+},
+{
+  cmd: 'stretchly mini -T "Stretch up !" --skip',
+  description: 'Skips to next Mini Break with "Stretch up!" title'
+},
+{
+  cmd: 'stretchly long -T "Stretch up !" -t "Go stretch !"',
+  description: 'Set next long break title to "Stretch up !" and text to "Go stretch !"'
+}]
+
+// Parse cmd line, check if valid and put variables in a dedicated object
+class Command {
+  constructor (input, version) {
+    this.version = version
+    this.supported = allCommands
+
+    this.parse(input)
+  }
+
+  parse (input) {
+    // filter out electron flags first
+    let i = 0
+    while (input[i].startsWith('--')) {
+      i++
+    }
+
+    const args = input.slice(i)
+    this.command = args[0]
+
+    if (this.command === undefined) {
+      this.command = 'help'
+    }
+
+    if (!this.supported[this.command]) {
+      throw new Error(`Error: command ${this.command} is not supported`)
+    }
+
+    this.options = this.getOpts(args.slice(1))
+  }
+
+  getOpts (opts) {
+    var options = {}
+
+    if (!this.supported[this.command].options) {
+      return null
+    }
+
+    for (let i = 0; i < opts.length; i++) {
+      const name = opts[i]
+      var valid = false
+
+      this.supported[this.command].options.forEach(opt => {
+        if (opt.long === name || opt.short === name) {
+          valid = true
+          if (opt.withValue) {
+            options[opt.long.slice(2)] = opts[i + 1]
+            i++
+          } else {
+            options[opt.long.slice(2)] = true
+          }
+        }
+      })
+
+      if (!valid) {
+        log.error(`Error: options ${name} is not valid for command ${this.command}`)
+      }
+    }
+
+    return options
+  }
+
+  runOrForward () {
+    if (!this.command) {
+      console.error('Error: command is undefined')
+      return
+    }
+
+    if (!this.supported[this.command]) {
```

Comment:
I think this will never happen as we catch this in `parse`. Let's use error there instead of throwing?

Reply:
Fixed by https://github.com/hovancik/stretchly/pull/700/commits/c82e0d097aa1ab4d4a8c7ba6d60c2d07d7d7c223